virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -711,3 +705,33 @@ def inflict_natural_mortality(
         )
 
         self.die_individual(number_of_deaths, carcass_pool)
+
+    def migrate_juvenile_probability(self) -> float:
+        """The probability that a juvenile cohort will migrate to a new grid cell.
+
+        TODO: This does not hold for diagonal moves.
+
+        Following Madingley's assumption that the probability of juvenile dispersal is
+        equal to the proportion of the cohort individuals that would arrive in the
+        neighboring cell after one full timestep's movement.
+
+        Assuming cohort individuals are homogenously distributed within a grid cell and
+        that the move is non-diagonal, the probability is then equal to the ratio of
+        dispersal speed to the side-length of a grid cell.
+
+        Returns:
+            The probability of diffusive natal dispersal to a neighboring grid cell.
+
+        """
+
+        A_cell = 1.0  #
+        grid_side = sqrt(A_cell)
+        velocity = sf.juvenile_dispersal_speed(
+            self.mass_current,
+            self.constants.V_disp,
+            self.constants.M_disp_ref,
+            self.constants.o_disp,
+        )
+        probability_of_dispersal = velocity / grid_side
```

Comment:
It will be useful to do a dimensional discussion here (an explanation is ok) to confirm this is unitless - as any probability should be - as well as a check to ensure that it is always <=1 .